src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataSerializerHelper.cs

```diff
@@ -0,0 +1,116 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="ODataSerializerHelper.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.AspNet.OData.Builder;
+using Microsoft.OData;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.AspNet.OData.Formatter.Serialization
+{
+    /// <summary>
+    /// Helper class for OData Serialization
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1053:StaticHolderTypesShouldNotHaveConstructors")]
+    internal class ODataSerializerHelper
+    {
+        /// <summary>
+        /// Appends instance annotations to an ODataResource.
+        /// </summary>
+        /// <param name="resource">The ODataResource being annotated</param>
+        /// <param name="resourceContext">The context for the resource instance to be annotated.</param>
+        /// <param name="value">The annotations to write.</param>
+        /// <param name="SerializerProvider">The SerializerProvider to use to write annotations</param>
+        internal static void AppendInstanceAnnotations(ODataResourceBase resource, ResourceContext resourceContext, object value, ODataSerializerProvider SerializerProvider)
+        {
+            IODataInstanceAnnotationContainer instanceAnnotationContainer = value as IODataInstanceAnnotationContainer;
+
```

Comment:
You could get rid of one level of nesting by returning early at this point if `instanceAnnotationContainer` is null.
```csharp
if (instanceAnnotationContainer == null)
{
    return;
}